### PR TITLE
Move SSH code into a common util package.  Add support for SSH tunnels.

### PR DESCRIPTION
**Pull request by brendandburns**
@cjcullen @mbforbes @lavalamp @thockin @roberthbailey 

This is the first part.  Tested with a simple test executable.  Unit tests seem vaguely feasible but I'd rather not, at least not initially.  TODO'd.

### CODE REVIEW

File: pkg/util/ssh.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
```
Comment (roberthbailey) [inline pkg/util/ssh.go]: nit: 2015

**Comment by lavalamp** [comment]
LGTM if e2e still passes

File: pkg/util/ssh.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+		SSHPort:    22,
+		LocalPort:  localPort,
+		RemotePort: remotePort,
+		RemoteHost: remoteHost,
+	}, nil
+}
+
+func (s *SSHTunnel) Open() error {
+	var err error
+	s.client, err = ssh.Dial("tcp", fmt.Sprintf("%s:%d", s.Host, s.SSHPort), s.Config)
```
Comment (cjcullen) [inline pkg/util/ssh.go]: Should these be net.JoinHostPort()?

Won't matter in our case, but might be nice if any of those IPv6ers want to use this util package.

Comment (cjcullen) [inline pkg/util/ssh.go]: feel free to ignore/TODO

File: pkg/util/ssh.go (hunk 1 above)
Comment (brendandburns) [inline pkg/util/ssh.go]: done.

File: pkg/util/ssh.go (hunk 2 above)
Comment (brendandburns) [inline pkg/util/ssh.go]: choosing to ignore for now...

**Comment by brendandburns** [comment]
e2e passes confirmed with: ```hack/ginkgo-e2e.sh --ginkgo.focus=SSH```

Ignoring the host/port thing (sorry @cjcullen )

**Comment by cjcullen** [comment]
LGTM (not that my comments matter ;) )

File: pkg/util/ssh.go (hunk 2 above)
Comment (lavalamp) [inline pkg/util/ssh.go]: eh... I'm with CJ on this one...

**Comment by brendandburns** [comment]
I did a bunch of re-factor in the follow on PR (https://github.com/GoogleCloudPlatform/kubernetes/pull/8908) and I converted it to strings there.  Can we merge this in order to avoid merge conflicts when I rebase the next PR?

**Comment by cjcullen** [comment]
I'm okay with this PR how it is.

**Pull request merged**